caluma/caluma_form/models.py

```diff
@@ -208,8 +208,48 @@ class Document(UUIDModel):
     form = models.ForeignKey(
         "caluma_form.Form", on_delete=models.DO_NOTHING, related_name="documents"
     )
+    source = models.ForeignKey(
+        "self",
+        blank=True,
+        null=True,
+        help_text="Reference this document has been copied from",
+        related_name="copies",
+        on_delete=models.SET_NULL,
+    )
     meta = JSONField(default=dict)
 
+    def copy(self, family=None):
+        """Create a copy including all its answers."""
+
+        new_document = self.__class__.objects.create(
```

Comment:
Nitpick: A cleaner way would be to use `type(self)` instead of `self.__class__`.

But feel free to change or merge nonetheless

Reply:
Done